internal/pkg/pkg.go

```diff
@@ -695,3 +699,49 @@ func SetPkgPathAnnotation(rn *yaml.RNode, pkgPath types.UniquePath) error {
 func RemovePkgPathAnnotation(rn *yaml.RNode) error {
 	return rn.PipeE(yaml.ClearAnnotation(pkgPathAnnotation))
 }
+
+// RGFile returns the resourcegroup object by lazy loading it from the filesytem.
+func (p *Pkg) RGFile(filename string) (*rgfilev1alpha1.ResourceGroup, error) {
+	if p.rgFile == nil {
+		// TODO(rquitales): Handle real reading errors vs file does not exist.
+		rg, _ := ReadRGFile(p.UniquePath.String(), filename)
```

Comment:
at-least start returning error and add a TODO to handling `NoExisting(...)`. Start strict at the beginning :)

Reply:
Error handling done.